contentful/marketplace-partner-apps
# Add Uploadcare app

## Purpose

Hey! We are Uploadcare and we want to add this integration to Contentful Marketplace.

We contacted with Matt Middleton and they said that first we should send a PR with out app for review.

## Approach

It seems like there is the only way for us to integrate with Contentful — “bind” our File Uploader to Contentful JSON object field. It is the same way you folks did together with Cloudinary, Bynder, etc.

## Testing steps

Here are the testing steps for the case where you server the app on localhost using dev server:

1. Create an account in Contentful. You also need to create a space there. If you have things like “Content types” and “Content” and you can create that things, you've done.
2. Clone the repo and install the deps for this app.
3. Run start script from the /apps/uploadcare directory.
4. Go to your management view, open “Apps” and click on “Create app”.
5. Fill the “Name” (any), “Frontend” (the URL you got on p.3), “Locations” (App configuration screen, Entry field → JSON object).
6. Click “Save”.
7. Go back to your space, create a content type with any “JSON object” field.
8. Go to “Apps”, then “Custom apps” and install your app. Fill up the settings properly.
9. Finally, create content with your content type and check that the field works properly.

For the settings you can use `demopublickey` as a API key or sign up for the free account on [uploadcare.com](https://uploadcare.com/). Any other setting is up to you.

To ensure that everything works properly you may upload any images to the field you've bind to Uploadcare app. You should be able to load, sort, and delete them as you wish. 

## Breaking Changes

This is an initial version.

## Dependencies and/or References

I think you may ask Matt Middleton about this case 🤔 

## Deployment

I'm not quite sure what I should provide here. But you probably need this public install link: https://app.contentful.com/deeplink?link=apps&id=4CGnwYyJEd5vpgHOmRv576.
